### PR TITLE
Update iterBinarySearch

### PR DESCRIPTION
**Pull request by Crazy3lf**
I added test cases for binary and linear searches and I notice the `iterBinarySearch` function does not passes the test cases. I make some minor changes to the algorithm.

### CODE REVIEW

File: searches/binary_search.go [hunk 1]
```diff
@@ -18,12 +18,12 @@ func iterBinarySearch(array []int, target int, lowIndex int, highIndex int) int
 	startIndex := lowIndex
 	endIndex := highIndex
 	var mid int
-	for startIndex < endIndex {
-		mid = int(startIndex + (endIndex - startIndex))
+	for startIndex <= endIndex {
+		mid = int(startIndex + (endIndex-1)/2)
```
Comment (Mystigan) [inline searches/binary_search.go:22]: - If highIndex is len(arrayName)-1, then why subtract 1 from the endIndex?

File: searches/search_test.go [hunk 2]
```diff
@@ -27,6 +31,15 @@ func TestBinarySearch(t *testing.T) {
 	}
 }
 
+func TestIterBinarySearch(t *testing.T) {
+	for _, test := range searchTests {
+		actual := iterBinarySearch(test.data, test.key, 0, len(test.data)-1)
+		if actual != test.expected {
+			t.Errorf("test %s failed", test.name)
```
Comment (Mystigan) [inline searches/search_test.go:38]: Suggest providing input, expected and actual values in the error message.

**Review by Mystigan — CHANGES_REQUESTED** [review]

File: searches/binary_search.go (hunk 1 above)
Comment (Crazy3lf) [reply searches/binary_search.go]: You are right, my mistake.

**Review by Mystigan — APPROVED** [review]
LGTM.

**Comment by Mystigan** [comment]
Merging changes as lint failure is unrelated to this PR.
Thanks for your contribution.

**Comment by Crazy3lf** [comment]
I added more exhaustive tests and notice my implementation still have errors. Do I open new pull request?

**Comment by Mystigan** [comment]
> I added more exhaustive tests and notice my implementation still have errors. Do I open new pull request?

What do you mean errors?

**Comment by Crazy3lf** [comment]
my brackets was in the wrong place and I should have used
```go
mid = int(startIndex + (endIndex - startIndex)/2)
```
to avoid overflow as stated in #123.

**Comment by Mystigan** [comment]
I have reverted all the merged changes. You may open a new PR with required changes. 
@Crazy3lf , Apologies, I missed it in the review.

**Comment by Crazy3lf** [comment]
> 
> 
> I have reverted all the merged changes. You may open a new PR with required changes.
> @Crazy3lf , Apologies, I missed it in the review.

It was my mistake, this is my first ever pull request so I am fairly inexperience😅. Sorry for the trouble.

**Pull request merged**